FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift

```diff
@@ -82,32 +82,28 @@ final class MessageService {
                 isEmail: true
             )
 
-            let isDecryptError = decrypted.blocks.isAnyError
+            let isWrongPassPhraseError = decrypted.blocks.first(where: { (block) -> Bool in
+                guard let errorBlock = block.decryptErr, case .needPassphrase = errorBlock.error.type else {
+                    return false
+                }
+                return true
+            })
 
-            if isDecryptError {
+            if isWrongPassPhraseError != nil {
                 reject(MessageServiceError.wrongPassPhrase(rawMimeData, passPhrase))
             } else {
                 keys
                     .map { PassPhrase(value: passPhrase, longid: $0.longid) }
                     .forEach { self.passPhraseStorage.savePassPhrase(with: $0, inStorage: false) }
```

Comment:
Todo for later - have another look to streamline this. The returned errors are handled partly here, but also partly in processMessage. Plus also interaction with saving pass phrases should be reviewed (it should happen earlier on), as soon as we know that the pass phrase matches our keys, not after it successfully decrypted a message.

Reply:
> as soon as we know that the pass phrase matches our keys
What is a proper way to check this?

Reply:
I'll have a look later

Reply:
let's discuss here https://github.com/FlowCrypt/flowcrypt-ios/issues/362